### PR TITLE
Fix Paxos V2 prepare response serialization

### PR DESCRIPTION
**Pull request by aweisberg**
Patch by Ariel Weisberg; Reviewed by David Capwell for CASANDRA-19023

### CODE REVIEW

**Review by dcapwell — CHANGES_REQUESTED** [review]
this doesn't match trunk, so won't this break paxos v2?

File: src/java/org/apache/cassandra/service/paxos/PaxosPrepare.java [hunk 1]
```diff
@@ -1232,7 +1232,6 @@ public void serialize(Response response, DataOutputPlus out, int version) throws
         {
             if (response.isRejected())
             {
-                out.writeByte(0);
```
Comment (bdeggleston) [inline src/java/org/apache/cassandra/service/paxos/PaxosPrepare.java:1235]: Could we just move the contents of `serializeRejection` back here? I don't think it needs to be in a separate method, especially when we're serializing the success response inline. @dcapwell this bug only exists in the cep-15 branch

**Review by bdeggleston — CHANGES_REQUESTED** [review]

**Review by bdeggleston — APPROVED** [review]
+1

**Review by dcapwell — APPROVED** [review]

**Comment by aweisberg** [comment]
Found a bug in size calculation as well. Removed `serializedRejection` and put the serialization of `maybeConsensusMigratedAt` at the end so it doesn't need to be repeated for accept/reject.

**Review by bdeggleston — APPROVED** [review]
+1

**Pull request merged**